File: lesson07/app/src/main/java/br/odb/nehe/lesson07/GL2JNILib.java
package br.odb.nehe.lesson07;

import android.content.res.AssetManager;
import android.graphics.Bitmap;

class GL2JNILib {

    public static AssetManager assetManager;

    static {
        System.loadLibrary("NdkGlue");
    }

    /**
     * @param width  the current view width
     * @param height the current view height
     */
    public static native void init(int width, int height);

    public static native void step();

    public static native void tick();

    public static native void onDestroy();

    public static native void onCreate(AssetManager assetManager);

    public static native void setTexture(Bitmap bitmap);

    public static native void toggleFiltering();

    public static native void toggleLightningMode();

    public static native void speedUpYZRotation();

    public static native void speedDownYZRotation();

    public static native void speedUpXZRotation();

    public static native void speedDownXZRotation();

    public static native void reset();

    public static native void dragTo(float x, float y);
}

File: lesson07/app/src/main/java/br/odb/nehe/lesson07/GL2JNILib.kt
package br.odb.nehe.lesson07

import android.content.res.AssetManager
import android.graphics.Bitmap

internal object GL2JNILib {

	/**
	 * @param width  the current view width
	 * @param height the current view height
	 */
	@JvmStatic
	external fun init(width: Int, height: Int)

	@JvmStatic
	external fun step()
	external fun tick()
	external fun onDestroy()
	external fun onCreate(assetManager: AssetManager?)
	external fun setTexture(bitmap: Bitmap?)
	external fun toggleFiltering()
	external fun toggleLightningMode()
	external fun speedUpYZRotation()
	external fun speedDownYZRotation()
	external fun speedUpXZRotation()
	external fun speedDownXZRotation()
	external fun reset()
	external fun dragTo(x: Float, y: Float)

	init {
		System.loadLibrary("NdkGlue")
	}
}